lib/svg-sprite/config.js

```diff
@@ -359,27 +369,36 @@ module.exports = class SVGSpriterConfig {
             transforms = defaultShapeTransform;
         }
 
-        const result = [];
-
         if (!Array.isArray(transforms)) {
-            return result;
+            return [];
         }
 
+        const result = [];
+
         for (let transform of transforms) {
             if (isString(transform)) {
-                transform = JSON.parse(`{"${transform}":true}`);
+                transform = {
+                    [transform]: true
+                };
             } else if (isFunction(transform)) {
-                const custom = {};
-                custom.custom = transform;
-                transform = custom;
+                transform = {
+                    custom: transform
+                };
             }
 
-            if (isObject(transform)) {
-                for (const [transformer, value] of Object.entries(transform)) {
-                    if (value === true || isObject(value) || isFunction(value)) {
-                        result.push([transformer, value === true ? {} : value]);
-                        break;
-                    }
+            if (!isObject(transform)) {
+                continue;
+            }
+
+            for (const [transformer, value] of Object.entries(transform)) {
+                if (value === true) {
+                    result.push([transformer, {}]);
+                    break;
+                }
+
+                if (isObject(value) || isFunction(value)) {
+                    result.push([transformer, value]);
+                    break;
```

Comment:
There is no need to break here

Reply:
You mean something like this?

```diff
 lib/svg-sprite/config.js | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/lib/svg-sprite/config.js b/lib/svg-sprite/config.js
index 4e32fe7..e0c59c0 100644
--- a/lib/svg-sprite/config.js
+++ b/lib/svg-sprite/config.js
@@ -393,13 +393,11 @@ module.exports = class SVGSpriterConfig {
             for (const [transformer, value] of Object.entries(transform)) {
                 if (value === true) {
                     result.push([transformer, {}]);
-                    break;
-                }
-
-                if (isObject(value) || isFunction(value)) {
+                } else if (isObject(value) || isFunction(value)) {
                     result.push([transformer, value]);
-                    break;
                 }
+
+                break;
             }
         }
 
```

Reply:
No, i mean the last `break` is redundant. There is no code left below this :)

Reply:
This is needed.